### PR TITLE
Changed default behaviour of RandomRotate

### LINKED ISSUE #115: Stretched volumes after RandomRotate for uneven dimensionalities (3D)
## 🐛 Bug Report

When using the trandsforms function RandomRotate on volumes with uneven dimensionalities ( in this example 64x64x128) the output doesn't look rotated correctly and furthermore looks transformed in an unwanted way.
When the input in padded with zero voxels to create an input shape of 128x128x128 the transformation looks to be correct. 
 

## 🔬 How To Reproduce

1. See coding example. For p=0.0 no transformation should apply and p=1.0 a transformation with should apply.
2. the example object is set to touch the boundaries of the volume in two dimensions
3. check output of the transformation

### Version 1. p =0.0 no transformation
```python

import numpy as np
import torch
from torch_trandsforms import RandomRotate, Compose
from trimesh.voxel.base import VoxelGrid
from trimesh import Scene


transform = Compose([
        RandomRotate([180,180,180], sample_mode="nearest", p=0.0, nd=3, keys=["input"]),
        ])

dummy_cube = torch.zeros((1, 64, 64, 128))
dummy_cube[:, :,:,32:96] = 1

transformed_cube = transform(input=dummy_cube)['input'].numpy()

vol_colors = np.zeros((*transformed_cube[0].shape, 4))
vol_colors[:,:,:,:] = [0, 0, 1, 0.1]
displayable_volume = VoxelGrid(transformed_cube[0]).as_boxes(colors=vol_colors)

scene = Scene()
scene.add_geometry(displayable_volume)
scene.show()
```

### Version 2. p =1.0 with transformation and the bug
```python
import numpy as np
import torch
from torch_trandsforms import RandomRotate, Compose
from trimesh.voxel.base import VoxelGrid
from trimesh import Scene


transform = Compose([
        RandomRotate([180,180,180], sample_mode="nearest", p=1.0, nd=3, keys=["input"]),
        ])

dummy_cube = torch.zeros((1, 64, 64, 128))
dummy_cube[:, :,:,32:96] = 1

transformed_cube = transform(input=dummy_cube)['input'].numpy()

vol_colors = np.zeros((*transformed_cube[0].shape, 4))
vol_colors[:,:,:,:] = [0, 0, 1, 0.1]
displayable_volume = VoxelGrid(transformed_cube[0]).as_boxes(colors=vol_colors)

scene = Scene()
scene.add_geometry(displayable_volume)
scene.show()
```
### Version 3. p =1.0 with transformation and padding to work around the bug
```python
import numpy as np
import torch
from torch_trandsforms import RandomRotate, Compose
from trimesh.voxel.base import VoxelGrid
from trimesh import Scene


transform = Compose([
        RandomRotate([180,180,180], sample_mode="nearest", p=1.0, nd=3, keys=["input"]),
        ])

dummy_cube = torch.zeros((1, 64, 64, 128))
dummy_cube[:, :,:,32:96] = 1
# apply padding to create equal dimensions
dummy_cube = torch.nn.functional.pad(dummy_cube, (0,0,32, 32, 32, 32), mode='constant', value=0)

transformed_cube = transform(input=dummy_cube)['input'].numpy()

vol_colors = np.zeros((*transformed_cube[0].shape, 4))
vol_colors[:,:,:,:] = [0, 0, 1, 0.1]
displayable_volume = VoxelGrid(transformed_cube[0]).as_boxes(colors=vol_colors)

scene = Scene()
scene.add_geometry(displayable_volume)
scene.show()
```
### Environment

* OS: [e.g. Linux & macOS]
* Python version 3.11.11
* torch-trandsforms  0.4.0
* numpy              2.2.2
* torch              2.5.1
* trimesh            4.6.1

### Screenshots

Image
Image

## 📈 Expected behavior

Correct rotation potentially within the limits of the volume or user defined padding when relevant.

### PR DESCRIPTION
**Pull request by ohjerm**
Rotate now by default operates in square/cubic/etc space, but crops to original size. Options have been added for minpadding to contain all the information, or maxpadding to return a square/cubic/etc result. "nothing" option performs no additional adjustments, retaining previous behaviour.

## Description

This is a semi-breaking change for those using RandomRotate and expecting it to perform a regular affine transformation. Use mode='nothing' to do "nothing" adjustments to the input.

## Related Issue

[115](https://github.com/alexandrainst/torch-trandsforms/issues/115)

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/ohjerm/torch-trandsforms/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/ohjerm/torch-trandsforms/blob/master/CONTRIBUTING.md) guide.
- [x] I've updated the code style using `make codestyle`.
- [x] I've written tests for all new methods and classes that I created.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: TIMING.md [hunk 1]
```diff
@@ -8,23 +8,24 @@ Automatically generated with test_speed.py
...
-| GaussianNoise | 0.0140 | 0.0012 | 0.1360 | 0.0057 |
-| RandomFlip | 0.0002 | 0.0000 | 0.0087 | 0.0000 |
-| RandomRotate | 0.0434 | 0.0016 | 0.5589 | 0.0115 |
-| RandomPadding | 0.0007 | 0.0000 | 0.0155 | 0.0001 |
+| RandomRotate90 | 0.0006 | 0.0000 | 0.0137 | 0.0000 |
+| UniformNoise | 0.0191 | 0.0000 | 0.1320 | 0.0000 |
+| Normalize | 0.0031 | 0.0000 | 0.0227 | 0.0000 |
+| SaltAndPepperNoise | 0.0426 | 0.0011 | 0.3265 | 0.0068 |
+| AdditiveBetaNoise | 0.0538 | 0.0014 | 0.3501 | 0.0093 |
+| GaussianNoise | 0.0132 | 0.0012 | 0.1423 | 0.0057 |
```
Comment (eikekutz) [inline TIMING.md:24]: Just curious:  Why did the other values also change?

Comment (eikekutz) [reply TIMING.md:24]: Transformations look reasonable. (visual inspection)
The values change slightly. Maybe it would make sense to add the option of scaling it to the original values, inform the user about the changes (if any) or scale it by default. 

example:
Data org max: 1.0, min: 1.0
Target org max: 1.0, min: 1.0
Data max: 1.000000238418579, min: 0.0
Target max: 1.000000238418579, min: 0.0
playground_3d_nothing
playground_3d_crop
playground_3d_maxpad
playground_3d_minpad

Comment (eikekutz) [reply TIMING.md:24]: Also tested bad values for the "mode" argument which fails (as it should)
